monitor/collector/collector.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package collector
+
+import (
+	"github.com/apache/dubbo-go/metrics/impl"
+	"github.com/prometheus/client_golang/prometheus"
+	"sync"
+)
+
+const (
+	groupName = "dubbo"
+)
+
+type Exporter struct {
+	//URI    string
+	metrics map[string]*prometheus.Desc
+	mutex   sync.RWMutex
+}
+
+func newGlobalMetric(namespace string, metricName string, docString string, labels []string) *prometheus.Desc {
+	return prometheus.NewDesc(namespace+"_"+metricName, docString, labels, nil)
+}
+
+func NewExporter(namespace string) (*Exporter, error) {
+	return &Exporter{
+		metrics: map[string]*prometheus.Desc{
+			"dubbogo_compass_metric": newGlobalMetric(namespace, "compass_metric", "The description of compass_metric", []string{"method_name"}),
+		},
+	}, nil
+}
+
+func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
+	for _, m := range e.metrics {
+		ch <- m
+	}
+}
+
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
```

Comment:
the lock scope is so large, sir. in my opinion, u just need to use the read lock in the for-loop.